app/classifier/tasks/dropdown/index.cjsx

```diff
@@ -133,56 +158,69 @@ module.exports = React.createClass
     false
 
   clearRelated: (i) ->
-    selects = @props.task.selects
+    selects = @state.selects
     {id} = selects[i]
     relatedSelects = Object.keys(selects).filter (key) =>
       selects[key].condition is id
     for key in relatedSelects
       @onChangeSelect(key, null)
 
   render: ->
-    {selects} = @props.task
-    selectKeys = Object.keys(selects)
+    {selects} = @state
 
     <GenericTask question={@props.task.instruction} help={@props.task.help} required={@props.task.required}>
       <div>
-        {selectKeys.map (i) =>
+        {selects.map (select, i) =>
           options = @getOptions(i)
           disabled = @getDisabledAttribute(i)
-          <div key={selects[i].id}>
-            {if selects[i].title isnt @props.task.instruction
-              <div>{selects[i].title}</div>}
-            <Select
+          <div key={select.id}>
+            {if select.title isnt @props.task.instruction
+              <div>{select.title}</div>}
+            {if select.allowCreate
+              SelectComponent = Select.Creatable
+            else
+              SelectComponent = Select}
+            <SelectComponent
               options={options}
               onChange={@onChangeSelect.bind(@, i)}
-              value={@props.annotation.value[i]?.value}
+              value={selects[i].selected}
               disabled={disabled}
               placeholder={if disabled then "N/A"}
-              allowCreate={selects[i].allowCreate}
               noResultsText={if not options?.length then null}
-              addLabelText="Press enter for {label}..."
+              promptTextCreator={(label) -> "Press enter for #{label}…"}
               matchPos="start"
               matchProp="label"
-              ref="select-#{i}"
+              ref={(instance) =>
+                return unless instance
+                if instance.select
+                  @menus[i] = instance.select
+                else
+                  @menus[i] = instance
+                }
             />
           </div>
         }
       </div>
     </GenericTask>
 
-  onChangeSelect: (i, newValue) ->
+  onChangeSelect: (i, option) ->
+    {selects} = @state
+    newValue = option.value if option?
     value = @props.annotation.value
 
-    optionsValues = @getOptions(i)?.map (option) -> option.value
-    newIndex = optionsValues?.indexOf(newValue)
+    options = @getOptions(i)
```

Comment:
Changing the code back gives the same result for me: `newIndex` is set to 0 because the new option has been added to the beginning of the options array before the change handler is called.